backend/src/main/java/com/praksa/auction/repository/CategoryRepository.java

```diff
@@ -2,7 +2,13 @@
 
 import com.praksa.auction.model.Category;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
 
 
 public interface CategoryRepository extends JpaRepository<Category, Long> {
+    @Query(value = "SELECT s.id, s.name,COUNT(p.subcategory_id) FROM Subcategory s LEFT JOIN Product p ON(s.id = p.subcategory_id) where s.category_id=:category Group by s.name,s.id", nativeQuery = true)
+    List<Object[]> findSubcategoriesWithCount(@Param("category") long category);
```

Comment:
Do we really need list of Objects here ? Can we use list of Categories instead ?

Reply:
These are subcategories, not categories. I tried using dto as return type, however it did not work, it only worked with Object []. Do you have any suggestions how to improve this?